### ISSUE TITLE
Submission for iOS bootcamp

### ISSUE THREAD

**Issue by jay-w-opus** [zhongjiewu/RockTip#1]
/cc @codepathreview

Basic work done for the calculator with some additional features. Demo GIF included. Please take a look. Thanks!

(this application is for the Airbnb bootcamp)

**Comment by nesquena** [comment]
Zhongjie,

Looks good, this exercise is intended in part to give you an introduction to the general rhythm of this course. The course is entirely project-based with an app being assigned each week and then due the following week. Each project builds on the last to help each engineer to learn all practical iOS development and best practices as quickly as possible. We also do a code review for each submitted project once the program begins.

Great to see you were able to complete a few extension tasks as well. The extension tasks available on each project are often the most valuable learnings since they dive deeper into common real-world iOS use cases.

Your application will be passed along to the organizers for final admissions. You should hear back again soon. I'd encourage you to continue working on app extensions in the meantime. Let me know if you have any questions.

**Comment by chieger** [comment]
Hi Zhongjie,

Just checking in to see if your Pre-work project is ready for review, or if you are working on any updates.

Thanks,
Charlie
CodePath

**Comment by jay-w-opus** [comment]
@chieger Yes. This was last time's submission but i backed out the application due to team priorities. I think this is in good state for this time's submission

**Comment by chieger** [comment]
Nice work Zhongjie! I think the next stretch goal would be UI Polish. 😎

Your application will be passed along to the organizers for final admissions. You should hear back again soon. I'd encourage you to continue working on app extensions in the meantime. Let me know if you have any questions.

#### Code styling in Swift. You can find some code styling guides here:

[Ray Wenderlich Swift Style Guide](https://github.com/raywenderlich/swift-style-guide)
[Github Swift Style Guide](https://github.com/github/swift-style-guide)

#### In this project, we explored the following concepts:
- Views are created in Storyboard, Interface Builder, or programmatically, but they have the same goal: instantiate, initialize, and layout view objects. We use IBOutlets to give names to view objects, similar to giving unique ids to divs in HTML.
- We registered for touch events, which can be done programmatically or via IBActions.
- We explored NSUserDefaults, one of the four persistence strategies in iOS.
- View controllers have a set of methods that are called when it loads, appears, or disappears. These are called view controller lifecycle methods.

Do your views look good on iPhone 4, 5, and 6? We will cover in class how to use Auto Layout to robustly design your views for different screen sizes and OS versions.

After this assignment, you should understand the purpose of IBOutlets and IBActions as well as the basics of designing views and programmatically interacting with the views from the controller.

Charlie Hieger
CodePath
